File: backend/src/main/java/com/services/PecaService.java
package com.services;

import com.database.model.Peca;
import com.database.dao.PecaDao;
import com.dto.requests.PecaRequest;
import com.dto.response.PecaResponse;
import com.exception.NotFoundException;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

@Service
@RequiredArgsConstructor
public class PecaService {
    private final PecaDao pecaDao;
    public List<PecaResponse> listarPecas(){
        return pecaDao.findAll()
                .stream()
                .map(PecaResponse::new)
                .toList();
    }

    public Peca buscarPeca(Long idPeca) throws NotFoundException {
        return pecaDao.findById(idPeca)
                .orElseThrow(()-> new NotFoundException("Peca não existe"));
    }

    @Transactional
    public Peca cadastrarPeca(PecaRequest peca) throws NotFoundException {
        Peca p = Peca.builder()
                .nome(peca.nome())
                .descricao(peca.descricao())
                .build();
        pecaDao.save(p);
        return p;
    }
}

File: frontend/front/src/services/PecaService.ts
import { api } from "@/src/services/api"

export const pecaService = {
    getPeca: async (id:number) => {
      const response = await api.get('/v1/peca/'+id);
      return response.data;
    },

    postPeca: async(nome: String, descricao: String) => {
      const payload = {
        nome: nome,
        descricao: descricao
      };
      const response = await api.post('/v1/peca', payload);
      return response.data;
    },

    getPecas: async() => {
      const response = await api.get('/v1/peca')
      return response.data
    }
  };